pages/xss-filter-evasion-cheatsheet.md

```diff
@@ -849,6 +849,49 @@ Assuming `http://www.google.com/` is programmatically replaced with nothing). I
 
     <A HREF="http://www.google.com/ogle.com/">XSS</A>
 
+
+
+
+## Assisting XSS with HTTP Parameter Pollution
+
+Assume a content sharing flow on a web site is implemented as below. There is a "Content" page which includes some content provided by users and this page also includes a link to "Share" page which enables a user choose his/her favourite social sharing platform to share on it. Developers HTML attribute encoded "title" parameter in "Content" page to prevent from XSS but for some reasons they didn't URL encoded this parameter to prevent from HTTP Parameter Pollution. And also they thought as content_type's value is a constant and it will always be integer as a result of this they didn't encode or validate content_type in "Share" page.
```

Comment:
```suggestion
Assume a content sharing flow on a web site is implemented as below. There is a "Content" page which includes some content provided by users and this page also includes a link to "Share" page which enables a user choose their favorite social sharing platform to share it on. Developers HTML encoded the "title" parameter in the "Content" page to prevent against XSS but for some reasons they didn't URL encoded this parameter to prevent from HTTP Parameter Pollution. Finally they decide that since content_type's value is a constant and will always be integer, they didn't encode or validate the content_type in the "Share" page.
```